cpg-language-java/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/java/JavaLanguage.kt

```diff
@@ -57,18 +57,34 @@ open class JavaLanguage :
     override val conjunctiveOperators = listOf("&&")
     override val disjunctiveOperators = listOf("||")
 
+    /**
+     * See
+     * [Java Language Specification](https://docs.oracle.com/javase/specs/jls/se19/html/jls-4.html).
+     */
     @Transient
     @JsonIgnore
     override val simpleTypes =
         mapOf(
-            "boolean" to IntegerType("boolean", 1, this, NumericType.Modifier.SIGNED),
+            // Boolean Types:
+            // https://docs.oracle.com/javase/specs/jls/se19/html/jls-4.html#jls-4.2.5
+            "boolean" to BooleanType("boolean", 1, this),
+            "Boolean" to BooleanType("java.lang.Boolean", 1, this),
+            "java.lang.Boolean" to BooleanType("java.lang.Boolean", 1, this),
```

Comment:
This is debatable. It is actually an ObjectType, not a boolean type. however you can do a direct assignment and boolean comparison on it, so I would suggest to model it as a boolean type:
```java
Boolean b = false;
if(b) {
}
```

Reply:
I would be fine with it but I am conerned that when the variable of type `Boolean` is actually used as an object something is weird, like a call on the object. Would WrapperType<BooleanType> make sense in the future to differentiate. 
The Save thing would be to make it an `ObjectType` or we try having it as `BooleanType` and see if something breaks?

Reply:
Lets have it as boolean type then.

Reply:
Actually, all `NumericType`s (and thus `BooleanType`) inherit from `ObjectType`, so nothing should really go wrong here.